src/omnicore/rpc.cpp

```diff
@@ -968,87 +964,119 @@ Value gettradessince_MP(const Array& params, bool fHelp)
-            "\nAllows user to request active order information from the order book\n"
-            
-            "\nArguments:\n"
-            "1. property_id1            (int, required) amount owned to up on sale\n"
-            "2. property_id2         (int, optional) property owned to put up on sale\n"
-            
-            "\nResult:\n"
-            "[                (array of string)\n"
-            "  \"hash\"         (string) Transaction id\n"            
-            "  ,...\n"
-            "]\n"
-
-            "\nbExamples\n"
-            //+ HelpExampleCli("trade_MP", "50", "3", "500", "5" )
-            //+ HelpExampleRpc("trade_MP", "50", "3", "500", "5" )
-        );
-  return "\nNot Implemented";
-}
-
-Value gettradehistory_MP(const Array& params, bool fHelp)
+/* TODO: point for discussion - "trade history" is ambiguous; providing trade history for a
+ * pair is likely to be more useful than trade history for an address.  Regardless the existing
+ * function did neither (only showing open trades, not trade history) so has been rewritten.
+ *
+ * Perhaps consider renaming this to gettradehistoryforaddress_OMNI to match existing naming
+ * (eg getallbalancesforaddress_MP) then look at addition of new gettradehistoryforpair_OMNI
+ * call.
+ */
+Value gettradehistoryforaddress_OMNI(const Array& params, bool fHelp)
 {
-   if (fHelp || params.size() < 1)
+    if (fHelp || params.size() < 1 || params.size() > 3)
         throw runtime_error(
             "gettradehistory_MP\n"
-            "\nAllows user to retreive trade history for some address on the Metadex\n"
-            
+            "\nAllows user to retrieve MetaDEx trade history for the supplied address\n"
             "\nArguments:\n"
-            "1. address          (string, required) address to query history on\n"
-            "2. number            (int, optional ) number of trades to retreive\n"
-            "3. property_id         (int, optional) filter by propertyid on one side\n"
+            "1. address          (string, required) address to retrieve history for\n"
+            "2. count            (int, optional) number of trades to retrieve (default: 10)\n"
+            "3. propertyid       (int, optional) filter by propertyid transacted\n"
         );
 
-  string address = params[0].get_str();
-  unsigned int number_trades = (params.size() == 2 ? (unsigned int) params[1].get_int64() : 512);
-  unsigned int propertyIdFilter = 0;
+    Array response;
+
+    string address = params[0].get_str();
+    int64_t count = 10;
+    if (params.size() > 1) {
+        count = params[1].get_int64();
+        if (count < 1) // TODO: do we need a maximum here?
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid value for count parameter");
+    }
+    uint32_t propertyId = 0;
+    if (params.size() > 2) {
+        int64_t tmpPropertyId = params[2].get_int64();
+        if (tmpPropertyId < 1 || tmpPropertyId > 4294967295) // not safe to do conversion
```

Comment:
These may be handy: [rpcrequirements.h](https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/src/omnicore/rpcrequirements.h) + [rpcvalues.h](https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/src/omnicore/rpcvalues.h)

Examples can be found [here](https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/src/omnicore/rpctx.cpp#L509-L519).

If a check/type is missing, then please ping me. :)

I wanted to go through all RPC calls at some point, and update the leftovers once this PR is in. :)